### PR TITLE
Implement search functionality in QuestionRepo component 

### PR DESCRIPTION
**Pull request by JunHao247**
•	Added search input to QuestionRepo.tsx to allow users to search questions.
•	Updated backend API to support searching by title, category, and complexity.
•	Enhanced README.md with detailed instructions on how to use the search, sort and pagination endpoint
•	Made the highlightText function reusable across components.

### CODE REVIEW

**Comment by JunHao247** [comment]
@gmengqi you may merge your PR if all is okay, will resolve merge conflict here

File: frontend/src/pages/QuestionRepo.tsx [hunk 1]
```diff
@@ -101,6 +104,19 @@ const QuestionRepo = () => {
...
+              label="Search"
+              value={searchQuery}
+              onChange={(e) => setSearchQuery(e.target.value)}
+              size="small"
+              variant="outlined"
+              fullWidth
+            />
+          </Box>
+          
+          {questions.length > 0 ? (
```
Comment (JunHao247) [inline frontend/src/pages/QuestionRepo.tsx:119]: @CedricChia123  Added no questions found typography.

File: question-service/src/service/question-service.ts [hunk 2]
```diff
@@ -8,18 +8,31 @@ export async function createQuestion(questionData: any) {
...
 
+    // Build filter object
+    let filter: any = {};
+
+    if (search) {
+        const searchRegex = new RegExp(search, 'i'); // Case-insensitive search
+    
+        filter.$or = [
+          { title: { $regex: searchRegex } },
+          { category: { $regex: searchRegex } },
```
Comment (JunHao247) [inline question-service/src/service/question-service.ts:21]: @CedricChia123 am still filtering the topic in search query, will remove if not required

**Review by chiewyx — APPROVED** [review]
LGTM, but see whether @gmengqi got any comments

**Comment by gmengqi** [comment]
LGTMT!

**Review by gmengqi — APPROVED** [review]

**Pull request merged**